src/scraper/scraper.cpp

```diff
@@ -4699,6 +4852,30 @@ bool LoadBeaconListFromConvergedManifest(const ConvergedManifest& StructConverge
     }
 }
 
+std::vector<uint160> GetVerifiedBeaconIDs(const ConvergedManifest& StructConvergedManifest)
+{
+    std::vector<uint160> result;
+    ScraperPendingBeaconMap VerifiedBeaconMap;
+
+    const auto& iter = StructConvergedManifest.ConvergedManifestPartsMap.find("VerifiedBeacons");
+    if (iter != StructConvergedManifest.ConvergedManifestPartsMap.end())
+    {
+        CDataStream part(iter->second, SER_NETWORK, 1);
+
+        part >> VerifiedBeaconMap;
+
+        for (const auto& entry : VerifiedBeaconMap)
+        {
+            CPubKey PubKey;
+
+            CKeyID KeyID = PubKey.Parse(entry.first).GetID();
```

Comment:
The map is keyed by `CKeyID` (`uint160`). I need to go back and check... I think we can just use `KeyID.SetHex(entry.first)` here.

Reply:
I will go back and check, but I think we stringified the scraper side maps. That is one reason I renamed them. I was getting confused between the scraper versions and the ones in the beacon class.

Reply:
typedef std::map<std::string, ScraperPendingBeaconEntry> ScraperPendingBeaconMap;

and from BeaconConsensus GetConsensusBeaconList()...

    for (const auto& pending_beacon_pair : pending_beacons)
    {
        const CKeyID& key_id = pending_beacon_pair.first;
        const NN::PendingBeacon& pending_beacon = pending_beacon_pair.second;

        if (pending_beacon.m_timestamp >= max_time)
        {
            continue;
        }

        ScraperPendingBeaconEntry beaconentry;

        beaconentry.timestamp = pending_beacon.m_timestamp;
        beaconentry.cpid = pending_beacon.m_cpid.ToString();

        consensus.mPendingMap.emplace(key_id.ToString(), std::move(beaconentry));
    }

So on the scraper side, the pending beacon map is keyed by the string of the CPID.

Reply:
Or are you talking about the superblock/beacon side map here. Maybe I am confused...
Yep. You're right. I am confused. You can go directly to the uint160 representation by doing SetHex on the string representation. Duh.